api/v1beta1/hetznerbaremetalmachine_types.go

```diff
@@ -77,85 +77,102 @@ type HetznerBareMetalMachineSpec struct {
 	// +optional
 	HostSelector HostSelector `json:"hostSelector,omitempty"`
 
-	// SSHSpec gives a reference on the secret where SSH details are specified as well as ports for ssh.
+	// SSHSpec gives a reference on the secret where SSH details are specified as well as ports for SSH.
 	SSHSpec SSHSpec `json:"sshSpec,omitempty"`
 }
 
 // HostSelector specifies matching criteria for labels on BareMetalHosts.
 // This is used to limit the set of BareMetalHost objects considered for
 // claiming for a Machine.
 type HostSelector struct {
-	// Key/value pairs of labels that must exist on a chosen BareMetalHost
+	// MatchLabels defines the key/value pairs of labels that must exist on a chosen BareMetalHost.
 	// +optional
 	MatchLabels map[string]string `json:"matchLabels,omitempty"`
 
-	// Label match expressions that must be true on a chosen BareMetalHost
+	// MatchExpressions defines the label match expressions that must be true on a chosen BareMetalHost.
 	// +optional
 	MatchExpressions []HostSelectorRequirement `json:"matchExpressions,omitempty"`
 }
 
 // HostSelectorRequirement defines a requirement used for MatchExpressions to select host machines.
 type HostSelectorRequirement struct {
-	Key      string             `json:"key"`
+	// Key defines the key of the label that should be matched in the host object.
+	Key string `json:"key"`
+
+	// Operator defines the [selection operator](https://pkg.go.dev/k8s.io/apimachinery@v0.23.4/pkg/selection?utm_source=gopls#Operator).
```

Comment:
can we either remove the link or set it on something that is not depending on the version?